compiler/rustc_metadata/src/creader.rs

```diff
@@ -698,11 +698,14 @@ impl<'a, 'tcx> CrateLoader<'a, 'tcx> {
         Ok(unsafe { *load_symbol_from_dylib::<*const &[ProcMacro]>(path, &sym_name)? })
     }
 
+    fn only_generating_rlib(&mut self) -> bool {
+        !self.tcx.crate_types().iter().any(|ct| *ct != CrateType::Rlib)
```

Comment:
```suggestion
        self.tcx.crate_types().iter().all(|ct| *ct == CrateType::Rlib)
```

avoids double negation.